model/src/main/java/org/projectnessie/api/v2/params/GetReferenceParams.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.api.v2.params;
+
+import java.util.Objects;
+import java.util.StringJoiner;
+import javax.annotation.Nullable;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import org.eclipse.microprofile.openapi.annotations.parameters.Parameter;
+import org.immutables.builder.Builder;
+import org.projectnessie.api.params.FetchOption;
+import org.projectnessie.model.Validation;
+
+public class GetReferenceParams {
+
+  @Parameter(ref = "refPathParameter")
+  @PathParam("ref")
+  @NotNull
+  @Pattern(regexp = Validation.REF_NAME_REGEX, message = Validation.REF_NAME_MESSAGE)
+  private String refName;
+
+  @Parameter(
+      description =
+          "Specify how much information to be returned. Will fetch additional metadata for references if set to 'ALL'.\n\n"
+              + "A returned Branch instance will have the following information:\n\n"
+              + "- numCommitsAhead (number of commits ahead of the default branch)\n\n"
+              + "- numCommitsBehind (number of commits behind the default branch)\n\n"
+              + "- commitMetaOfHEAD (the commit metadata of the HEAD commit)\n\n"
+              + "- commonAncestorHash (the hash of the common ancestor in relation to the default branch).\n\n"
+              + "- numTotalCommits (the total number of commits in this reference).\n\n"
+              + "A returned Tag instance will only contain the 'commitMetaOfHEAD' and 'numTotalCommits' fields.\n\n"
+              + "Note that computing & fetching additional metadata might be computationally expensive on the server-side, so this flag should be used with care.")
+  @QueryParam("fetch")
+  @Nullable
+  private FetchOption fetchOption;
+
+  public GetReferenceParams() {}
+
+  @Builder.Constructor
```

Comment:
in other places we use `@org.immutables.builder.Builder.Constructor`

also makes me wonder (pardon my ignorance): why are we implementing tostring/equals ourselves?
was it not possible to use immutables because we need the openapi annotations?

Reply:
copy-paste in this case. I'll see if we can use the generated methods.

Reply:
Added imports for that annotation

Reply:
I believe the hand-written classes are required for JaxRs... While the spec is not very clear on that when method parameters are concerned, I tend to think it implies having public constructors in parameter classes too.

Reply:
As for tostring/equals I do not think we need them in param classes at all.

Reply:
I've removed equals/hash code in 470df9f9138415f68a26ea2544b7c4ea861eadea